core_lib/src/structure/layer.cpp

```diff
@@ -666,3 +666,24 @@ KeyFrame* Layer::getKeyFrameWhichCovers(int frameNumber) const
     }
     return nullptr;
 }
+
+QDomElement Layer::createBaseDomElement(QDomDocument& doc)
+{
+    QDomElement layerTag = doc.createElement("layer");
+    layerTag.setAttribute("id", id());
+    layerTag.setAttribute("name", name());
+    layerTag.setAttribute("visibility", visible());
+    layerTag.setAttribute("type", type());
+    return layerTag;
+}
+
+void Layer::loadBaseDomElement(QDomElement& elem)
+{
+    if (!elem.attribute("id").isNull())
+    {
+        int id = elem.attribute("id").toInt();
+        setId(id);
+    }
+    setName(elem.attribute("name", "untitled"));
+    setVisible(elem.attribute("visibility", "true").toInt());
```

Comment:
The default value should be convertible to an int.

Reply:
Fixed. I expected a stringified bool to become an int but after testing I see that, that was a bad assumption.